Use smart pointers in GraphicsContextGLCVCocoa

rwlbuis opened the pull request:
#### 8d4e52898d91b34ac71de6a8f259c8ee929ae396
<pre>
Use smart pointers in GraphicsContextGLCVCocoa
<a href="https://bugs.webkit.org/show_bug.cgi?id=295027">https://bugs.webkit.org/show_bug.cgi?id=295027</a>

Reviewed by NOBODY (OOPS!).

Fix remaining warnings of type [alpha.webkit.UnretainedCallArgsChecker].

* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/platform/graphics/cv/GraphicsContextGLCVCocoa.mm:
(WebCore::GraphicsContextGLCVCocoa::copyVideoSampleToTexture):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8d4e52898d91b34ac71de6a8f259c8ee929ae396

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119940 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39633 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30284 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126263 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72013 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121816 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40328 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48210 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91080 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60391 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122892 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32208 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107541 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71636 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31238 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25647 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69901 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101682 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25836 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129186 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46860 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35522 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99698 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47226 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103731 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99543 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25272 "") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44995 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23008 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43481 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46722 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52428 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46188 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49537 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47874 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/platform/graphics/cv/GraphicsContextGLCVCocoa.mm:
```diff
@@ -182,17 +182,17 @@ static TransferFunctionCV transferFunctionFromString(CFStringRef string)
     if (!string)
         return TransferFunctionCV::Unknown;
-    if (CFEqual(string, kCVImageBufferYCbCrMatrix_ITU_R_709_2))
+    if (RetainPtr matrix = kCVImageBufferYCbCrMatrix_ITU_R_709_2; CFEqual(string, matrix.get()))
```
We should just fix the static analyzer to treat CFEqual as a trivial function.

rwlbuis replied:
I made an attempt at https://github.com/llvm/llvm-project/pull/146369. Not sure if you are a reviewer. Also feel free to take over if it is way off. The most important thing is to get it fixed, and then I can rebase this PR and it would be a one-liner PR..

rwlbuis replied:
Now landed (https://github.com/llvm/llvm-project/commit/524f0903064c7ffd1a1f3b3b6f7d5da31151bfca).

rwlbuis replied:
I rebased, once the CFEqual fix is available on the smartcpp bot, this should be basically a one liner fix for GraphicsContextGLCVCocoa.

rwlbuis committed:
Use smart pointers in GraphicsContextGLCVCocoa
https://bugs.webkit.org/show_bug.cgi?id=295027

Reviewed by NOBODY (OOPS!).

Fix remaining warnings of type [alpha.webkit.UnretainedCallArgsChecker].

* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/platform/graphics/cv/GraphicsContextGLCVCocoa.mm:
(WebCore::GraphicsContextGLCVCocoa::copyVideoSampleToTexture):
